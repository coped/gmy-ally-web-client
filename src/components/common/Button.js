import React from "react";

export default function Button(props) {
  const classModifiers = props.classModifiers ? props.classModifiers : "";
  return (
    <button className={"button " + classModifiers} onClick={props.onClick}>
      {props.children}
    </button>
  );
}
