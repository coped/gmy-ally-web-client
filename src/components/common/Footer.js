import React from "react";
import logo from "images/logo_transparent.png";
import "assets/Footer.scss";

export default function Footer() {
  return (
    <div id="Footer">
      <footer className="footer">
        <div className="content has-text-centered">
          <p>
            Gym Partner is a work in progress! You can check out the source code
            and make contributions on{" "}
            <a href="https://github.com/coped/gym-partner-api">
              Gym Partner's github.
            </a>
          </p>
          <img src={logo} alt="Gym Partner logo" className="logo" />
        </div>
      </footer>
    </div>
  );
}
