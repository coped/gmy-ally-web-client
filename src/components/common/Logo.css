.Logo-animated {
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .Logo-animated {
    animation: Logo-breathe alternate infinite 1.5s;
    }
}

@keyframes Logo-breathe {
    from {
    transform: scale(1, 1);
    }
    to {
    transform: scale(1.2, 1.2);
    }
}