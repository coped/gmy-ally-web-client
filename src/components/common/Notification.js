import React from "react";

export default function Notification(props) {
  const { classList = [], children, type, ...rest } = props;
  const classes = ["notification", "is-medium", `is-${type}`, ...classList];
  return (
    <div className={classes.join(" ")} {...rest}>
      {children}
    </div>
  );
}
