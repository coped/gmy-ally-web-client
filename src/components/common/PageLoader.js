import React from "react";

export default function PageLoader({
  message = "Loading . . .",
  loading = false,
}) {
  const baseClasses = ["pageloader", "is-info"];
  const classes = loading ? [...baseClasses, "is-active"] : [...baseClasses];
  return (
    <div className={classes.join(" ")}>
      <span className="title">{message}</span>
    </div>
  );
}
