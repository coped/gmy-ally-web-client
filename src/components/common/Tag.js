import React from "react";

export default function Tag(props) {
  return (
    <div>
      <span className={`tag is-${props.type} is-medium`}>
        {props.message}
      </span>
    </div>
  );
}
