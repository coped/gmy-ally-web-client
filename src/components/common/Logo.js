import React from "react";
import "assets/images/Logo.css";
import gymPartnerLogo from "assets/images/logo_transparent.png";

export default function Logo(props) {
  return (
    <img
      src={gymPartnerLogo}
      className={props.isAnimated ? "Logo-animated" : ""}
      alt="logo"
    />
  );
}
