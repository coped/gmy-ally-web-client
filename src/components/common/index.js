import Button from "components/common/Button";
import Logo from "components/common/Logo";
import Notification from "components/common/Notification";

export { Button, Logo, Notification };
