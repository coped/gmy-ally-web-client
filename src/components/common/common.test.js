import React from "react";
import { render } from "@testing-library/react";
import { Button, Logo, TextInputField, Notification } from "components/common";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  const text = "I'm a button";
  const props = {
    classModifiers: "is-link",
    onClick: () => console.log("this is a button"),
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button>{text}</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders children correctly", () => {
    const wrapper = shallow(<Button>{text}</Button>);
    expect(wrapper.find("button").text()).toEqual(text);
  });

  it("renders without crashing with props", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button {...props}>I am a button</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(
      <Button {...props}>I am a button</Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Logo", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Logo />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Logo />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("TextInputField", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TextInputField />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<TextInputField />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Notification", () => {
  const props = { type: "info" };
  const message = "I'm a notification.";

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Notification>{message}</Notification>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("properly renders children", () => {
    const wrapper = shallow(
      <Notification {...props}>
        <p>{message}</p>
      </Notification>
    );
    expect(wrapper.find("p").text()).toEqual(message);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(
      <Notification {...props}>{message}</Notification>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
