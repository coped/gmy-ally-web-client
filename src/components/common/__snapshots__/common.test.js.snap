// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className="button is-link"
  onClick={[Function]}
>
  I am a button
</button>
`;

exports[`Logo has a valid snapshot 1`] = `
<img
  alt="logo"
  className="Logo-unanimated"
  src="logo_transparent.png"
/>
`;

exports[`Notification has a valid snapshot 1`] = `
<div
  className="notification is-info is-medium"
>
  I'm a notification.
</div>
`;
